cmd/ingress-operator/start.go

```diff
@@ -207,6 +207,18 @@ func start(opts *StartOptions) error {
 // createDNSManager creates a DNS manager compatible with the given cluster
 // configuration.
 func createDNSProvider(cl client.Client, operatorConfig operatorconfig.Config, dnsConfig *configv1.DNS, platformStatus *configv1.PlatformStatus) (dns.Provider, error) {
+	// If no DNS configuration is provided, don't try to set up provider clients.
+	// TODO: the provider configuration can be refactored into the provider
```

Comment:
It confused me for a moment to see the TODO added in this context.  The TODO really applies to the logic in the `switch` below, and not specifically to this new short-circuiting logic, right?

Reply:
well, the short-circuiting logic would go away in the refactor I'm imagining, so maybe the TODO applies to the entire function?